crates/oxc_codegen/src/lib.rs

```diff
@@ -412,11 +412,13 @@ impl<'a> Codegen<'a> {
     }
 
     fn print_list<T: Gen>(&mut self, items: &[T], ctx: Context) {
-        for (index, item) in items.iter().enumerate() {
-            if index != 0 {
-                self.print_comma();
-                self.print_soft_space();
-            }
+        let mut iter = items.iter();
+        if let Some(item) = iter.next() {
+            item.gen(self, ctx);
+        }
```

Comment:
No need to fall through to the `while` loop if `iter` is empty.

```suggestion
        let Some(item) = iter.next() else { return; };
        item.gen(self, ctx);
```